sdk/src/types/block/payload/signed_transaction/transaction.rs

```diff
@@ -111,18 +98,28 @@ impl RegularTransactionEssenceBuilder {
         self
     }
 
-    /// Adds a payload to a [`RegularTransactionEssenceBuilder`].
+    /// Adds a payload to a [`TransactionBuilder`].
     pub fn with_payload(mut self, payload: impl Into<OptionalPayload>) -> Self {
         self.payload = payload.into();
         self
     }
 
-    /// Finishes a [`RegularTransactionEssenceBuilder`] into a [`RegularTransactionEssence`].
-    pub fn finish_with_params<'a>(
-        self,
-        params: impl Into<ValidationParams<'a>> + Send,
-    ) -> Result<RegularTransactionEssence, Error> {
+    /// Adds outputs to a [`TransactionBuilder`].
```

Comment:
this would replace any outputs being added through `add_output`, so here as I well I'ld say "Sets ..."

Reply:
Reviewing older commit, has been changed already.